src/Bicep.LangServer/Handlers/BicepBuildCommandHandler.cs

```diff
@@ -0,0 +1,125 @@
+        {
+            if (string.IsNullOrWhiteSpace(bicepFilePath))
+            {
+                throw new ArgumentException("Invalid input file");
+            }
+
+            DocumentUri documentUri = DocumentUri.FromFileSystemPath(bicepFilePath);
+            string buildOutput = GenerateCompiledFileAndReturnBuildOutputMessage(bicepFilePath, documentUri);
+
+            return Task.FromResult(buildOutput);
+        }
+
+        private string GenerateCompiledFileAndReturnBuildOutputMessage(string bicepFilePath, DocumentUri documentUri)
+        {
+            StringBuilder sb = new StringBuilder();
+
+            string? compiledFilePath = GetCompiledFilePath(bicepFilePath, sb, out string bicepFile, out string compiledFile);
+
+            if (string.IsNullOrWhiteSpace(compiledFilePath))
+            {
+                throw new ArgumentException($"Invalid input file");
+            }
+
+            CompilationContext? context = CompilationManager.GetCompilation(documentUri);
+
+            if (context is null)
+            {
+                throw new InvalidOperationException($"Unable to get compilation context");
+            }
+
+            SemanticModel semanticModel = context.Compilation.GetEntrypointSemanticModel();
+            KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile = context.Compilation.GetAllDiagnosticsByBicepFile()
+                .FirstOrDefault(x => x.Key.FileUri == documentUri.ToUri());
+
+            if (diagnosticsByFile.Value.Any())
+            {
+                AppendDiagnosticsMessage(diagnosticsByFile, bicepFile, sb);
+                return sb.ToString();
+            }
+
+            using (FileStream fileStream = new FileStream(compiledFilePath, FileMode.Create, FileAccess.ReadWrite, FileShare.ReadWrite))
+            {
+                TemplateEmitter emitter = new TemplateEmitter(semanticModel, ThisAssembly.AssemblyFileVersion);
+                EmitResult result = emitter.Emit(fileStream);
+
+                sb.AppendLine("Build succeeded. Created transpiled ARM template: " + compiledFile);
+            }
+
+            return sb.ToString();
+        }
+
+        private void AppendDiagnosticsMessage(KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile, string bicepFile, StringBuilder sb)
+        {
+            sb.AppendLine("Build failed. Please fix below errors in " + bicepFile + " :");
+
+            IEnumerable<IDiagnostic> diagnostics = diagnosticsByFile.Value;
+            IReadOnlyList<int> lineStarts = diagnosticsByFile.Key.LineStarts;
+
+            for (int i = 0; i < diagnostics.Count(); i++)
+            {
+                IDiagnostic diagnostic = diagnostics.ElementAt(i);
+                var (startLine, startChar) = TextCoordinateConverter.GetPosition(lineStarts, diagnostic.Span.Position);
+
+                sb.AppendLine((i+1) + ". " + diagnostic.Message + " bicep(" + diagnostic.Code + "). " + "[" + (startLine + 1) + ", " + (startChar + 1) + "]");
```

Comment:
>AppendLine [](http://example.com/codeflow?start=19&length=10)

Can we try the same exact format that the CLI returns? It's a standard format used by tsc, msbuild, and other tools like that. 

VS code runs "problem matchers" (not sure what the official name is) on the output and turns errors into clickable links. I'm wondering if we could get the same behavior by matching the output format.

Reply:
Thanks for the feedback. Didn't know about the problem matchers. Updated here:
https://github.com/Azure/bicep/commit/afb789d670fba3182e8f077c1da6e8c5cd5483fd